Userland/Services/AudioServer/main.cpp

```diff
@@ -31,7 +31,7 @@ ErrorOr<int> serenity_main(Main::Arguments)
         (void)IPC::new_client_connection<AudioServer::ClientConnection>(move(client_socket), client_id, *mixer);
     };
 
-    TRY(Core::System::pledge("stdio recvfd thread accept cpath rpath wpath"));
+    TRY(Core::System::revoke("unix"));
```

Comment:
Taking this a bit further, if we want to have `revoke()` why not just remove `pledge()` and force the caller to `revoke()` everything instead? You switch from saying everything a process can do, to everything it can't...

I think it's much more difficult to keep it in your head this way, but it seems like it's worth bringing up if we are discussing this 😀.

Reply:
This is exactly my point in #11140: force `LibMain` users to provide their initial promises via a function and let `LibMain` set these up. Allow applications to only `revoke()` things afterwards.

Reply:
Why though? What do we gain by adding extra complexity? I don't see the value at the moment.

Reply:
This forces every SerenityOS application to `pledge()`, which seems beneficial. It makes promises fully DRY. It simplifies dropping promises during runtime and makes it easy to spot where and which privileges are dropped in code and backtraces.

I see it as removing code complexity by compile-time `pledge()` enforcement and removing duplicate promise definitions. This uncovered a real bug already.

Reply:
From #11140: after #11207 and #11211 I consider the repeated `pledge()` idiom actively harmful for security audits. I've removed duplicate (redundant) pledges from three programs now, but I have no clue if these point to bugs in the security or that they're merely copy-and-paste errors.

If applications would `pledge()` once and then `revoke()` it signals clear programmer intent.